src/main/java/com/google/cloud/teleport/templates/PubsubToText.java

```diff
@@ -166,11 +173,40 @@ public static PipelineResult run(Options options) {
                         options.getOutputShardTemplate(),
                         options.getOutputFilenameSuffix()))
                 .withTempDirectory(NestedValueProvider.of(
-                    options.getOutputDirectory(),
+                    maybeUseUserTempLocation(
+                        options.getUserTempLocation(),
+                        options.getOutputDirectory()),
                     (SerializableFunction<String, ResourceId>) input ->
                         FileBasedSink.convertToFileResourceIfPossible(input))));
 
     // Execute the pipeline and return the result.
     return pipeline.run();
   }
+
+  /**
+   * Utility method for using optional parameter userTempLocation as TempDirectory.
+   * This is useful when output bucket is locked and temporary data cannot be deleted.
+   *
+   * @param userTempLocation user provided temp location
+   * @param outputLocation default temp location, which should OutputDirectory for backward compatibility
```

Comment:
Maybe change this to:
"user provided outputDirectory to be used as the default temp location"

Reply:
👍